Add CLI support to generate morphometrics file. (#434)
neuropoly/axondeepseg: docs/source/documentation.rst

    axondeepseg -t SEM -i test_segmentation/test_sem_image/image1_sem/

Segment images from multiple folders
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To segment images that are located in different folders, specify the path to the folders in the **-i** argument, one after the other. For instance, to segment all the images of folders **'test_sem_image/image1_sem/'** and **'test_sem_image/image2_sem/'** of the test dataset, use the following command::

    axondeepseg -t SEM -i test_segmentation/test_sem_image/image1_sem/ test_segmentation/test_sem_image/image2_sem/

Morphometrics

> ---------------

Reviewer: ```suggestion
-------------
```

You can generate morphometrics using AxonDeepSeg via the command line interface.

Syntax

> ~~~~~

Reviewer: ```suggestion
~~~~~~
```

    axondeepseg -t SEM -i test_segmentation/test_sem_image/image1_sem/

Segment images from multiple folders
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To segment images that are located in different folders, specify the path to the folders in the **-i** argument, one after the other. For instance, to segment all the images of folders **'test_sem_image/image1_sem/'** and **'test_sem_image/image2_sem/'** of the test dataset, use the following command::

    axondeepseg -t SEM -i test_segmentation/test_sem_image/image1_sem/ test_segmentation/test_sem_image/image2_sem/

Morphometrics
---------------

You can generate morphometrics using AxonDeepSeg via the command line interface.

Syntax
~~~~~

The script to launch in called **axondeepseg_morphometrics**. It has several arguments.

**Required arguments:**

-i IMGPATH
                    Path to the image file whose morphometrics needs to be calculated.

**Optional arguments:**

-s SIZEPIXEL        Pixel size of the image(s) to segment, in micrometers. 

>                     If no pixel size is specified, a **pixel_size_in_micrometer.txt** file needs to be added to the image folder path ( that file should contain a single float number corresponding to the resolution of the image, i.e. the pixel size). The pixel size in that file will be used for the segmentation.

Reviewer: ```suggestion
                    If no pixel size is specified, a **pixel_size_in_micrometer.txt** file needs to be added to the image folder path (that file should contain a single float number corresponding to the resolution of the image, i.e. the pixel size). The pixel size in that file will be used for the morphometrics computation.
```

-a AXONSHAPE       Axon shape
                    **circle:** Axon shape is considered as circle. In this case, diameter is computed using equivalent diameter. 
                    **ellipse:** Axon shape is considered as an ellipse. In this case, diameter is computed using ellipse minor axis.
                    The default axon shape is set to **circle**.

> -f FILENAME         Name of the excel file in which the morphometrics file will be stored.

Reviewer: ```suggestion
-f FILENAME         Name of the excel file in which the morphometrics will be stored.
```

>                     The excel file extension can either be **.xlsx** or **.csv**.

Reviewer: The `csv` option is not working, see comment in `launch_morphometrics_computation.py`.

Author: Fixed this.

                    If name of the excel file is not provided, the morphometrics will be saved as **morphometrics.xlsx**.

Morphometrics of a single image
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Before computing the morphometrics of an image, make sure it has been segmented using AxonDeepSeg ::

    axondeepseg_morphometrics -i test_segmentation/test_sem_image/image1_sem/77.png -a circle -f morphometrics 

This generates a morphometrics file in the image directory::

    --image1_sem/
    ---- 77.png
    ---- 77_seg-axon.png
    ---- 77_seg-axonmyelin.png
    ---- 77_seg-myelin.png
    ---- pixel_size_in_micrometer.txt
    ---- morphometrics.xlsx
    ...

.. NOTE :: By default, AxonDeepSeg treats axon shape as **circle** and the calculation of the diameter is based on the axon area of the mask. 

>            For each axons, the equivalent diameter is computed, which is the diameter of a circle with the same area as the axon ::

Reviewer: ```suggestion
           For each axons, the equivalent diameter is computed, which is the diameter of a circle with the same area as the axon. ::
```

           
           If you wish to treat axon shape as an ellipse, you can set the  **-a** argument to be **ellipse**.
           When axon shape is set to ellipse, the calculation of the diameter is based on ellipse minor axis::
            
            axondeepseg -i test_segmentation/test_sem_image/image1_sem/77.png -a ellipse

Morphometrics of images from multiple folders
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
To generate morphometrics of images which are located in different folders, specify the path of the image folders using the **-i** argumument of the CLI separated by space. For instance, to compute morphometrics of the images present in the folders **'test_sem_image/image1_sem/'** and **'test_sem_image/image2_sem/'** of the test dataset, use the following command::

>     axondeepseg_morphometrics -i test_sem_image/image1_sem/77.png test_sem_image/image2_sem/image.png

Reviewer: ```suggestion
    axondeepseg_morphometrics -i test_segmentation/test_sem_image/image1_sem/77.png test_segmentation/test_sem_image/image2_sem/image.png
```

    --image1_sem/
    ---- 77.png
    ---- 77_seg-axon.png
    ---- 77_seg-axonmyelin.png
    ---- 77_seg-myelin.png
    ---- pixel_size_in_micrometer.txt
    ---- morphometrics.xlsx
    ...

    --image2_sem/
    ---- image.png
    ---- image_seg-axon.png
    ---- image_seg-axonmyelin.png
    ---- image_seg-myelin.png
    ---- pixel_size_in_micrometer.txt
    ---- morphometrics.xlsx

Axon Shape: Circle vs Ellipse
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Circle 
^^^^^^
**Usage** ::

    axondeepseg -i test_segmentation/test_sem_image/image1_sem/77.png -a circle

**Studies using Circle as axon shape:**

* Duval et al: https://pubmed.ncbi.nlm.nih.gov/30326296/
* Salini et al: https://www.frontiersin.org/articles/10.3389/fnana.2017.00129/full

Ellipse
^^^^^^^
**Usage** ::

    axondeepseg -i test_segmentation/test_sem_image/image1_sem/77.png -a ellipse

**Studies using Ellipse as axon shape:**

* Payne et al: https://pubmed.ncbi.nlm.nih.gov/21381867/
* Payne et al: https://pubmed.ncbi.nlm.nih.gov/22879411/
* Fehily et al: https://pubmed.ncbi.nlm.nih.gov/30702755/

.. NOTE :: In the literature, both equivalent diameter and ellipse minor axis are used to compute the morphometrics. 

>            Thus, depending on the usecase the user is advised to choose axon shape accordingly.

Reviewer: ```suggestion
           Thus, depending on the usecase, the user is advised to choose axon shape accordingly.
```
